src/viking.js

```diff
@@ -1,14 +1,113 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength){
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack(){
+    return this.strength
+  }
+
+  receiveDamage(damage){
+    this.health -= damage
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+
+  constructor(name,health,strength) {
+    super(health, strength)
+    this.name = name;
+  }
+
+  receiveDamage(damage){
+    this.health -= damage
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`
+    } else {
+      return `${this.name} has died in act of combat` 
+    }
+  }
+
+  battleCry(){
+    return `Odin Owns You All!`
+  }
+
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  
+  receiveDamage(damage){
+    this.health -= damage
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`
+    } else {
+      return `A Saxon has died in combat` 
+    }
+  }
+
+}
 
 // War
-class War {}
+class War {
+
+  
+  constructor(){
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+
+  
+  addViking(newViking){
+    this.vikingArmy.push(newViking)
+    
+  }
+  
+  
+  addSaxon(newSaxon){
+    this.saxonArmy.push(newSaxon)
+  }
+
+
+  vikingAttack(){
+    const randomNumberSaxon = Math.floor(Math.random() * this.saxonArmy.length)
+    const randomNumberViking = Math.floor(Math.random() * this.vikingArmy.length)
+    const result = this.saxonArmy[randomNumberSaxon].receiveDamage(this.vikingArmy[randomNumberViking].strength)
+    if (this.saxonArmy[randomNumberSaxon].health <= 0 ) {
+      this.saxonArmy.splice([randomNumberSaxon])
+    }
+    return result 
+ 
+  }
```

Comment:
Está bien lo que has implementado, pero con un filter también se podría realizar. Te lo pongo a continuación y así tienes otra manera de verlo!!
`   vikingAttack() {
    const viking = randomElement(this.vikingArmy);
    const saxon = randomElement(this.saxonArmy);

    const attackResult = saxon.receiveDamage(viking.strength);
    this.saxonArmy = this.saxonArmy.filter(saxon => saxon.health > 0);
    return attackResult;
  }`